remote/remote.go

```diff
@@ -194,11 +194,13 @@ func newV1Image(keychain authn.Keychain, repoName string, platform imgutil.Platf
 		image, err = remote.Image(ref, remote.WithAuth(auth), remote.WithTransport(http.DefaultTransport), remote.WithPlatform(v1Platform))
 		if err != nil {
 			if err == io.EOF && i != maxRetries {
```

Comment:
It doesn't relate to your commit but I've just realized that `maxRetries` is set to `2` and the total number of retries is `3` because we start from `0`.
Maybe it's better to set `maxRetries` to 3 and change the `for loop` to be:
```
for i := 0; i < maxRetries; i++ {
```
I think it'll be clearer for the reader.